src/lib/wasm/MultipassStore.ts

```diff
@@ -59,7 +59,20 @@ class MultipassStore {
                 case wasm.MultiPassEventKindEnum.FriendRequestReceived: {
                     if (get(SettingsStore.state).notifications.friends) {
                         let incoming = await this.identity_from_did(event.did)
-                        Store.addToastNotification(new ToastMessage("New friend request.", `${incoming?.name} sent a request.`, 2), Sounds.Notification)
+                        let count = 0
+                        while (incoming === undefined) {
+                            incoming = await this.identity_from_did(event.did)
+                            count++
+                            if (count > 15) {
+                                break;
+                            }
+                            await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
so does this retry for 15 times once per second? Maybe we should move these values into `src/lib/config.ts` as "MAX_RETRY" and "RETRY_TIMEOUT" so that we can easily update the same logic elsewhere. 

Small other note to remove extra `;`. I think this method overall is fine but perhaps we can build a retry util in polling.ts to try again in 500ms then wait 1000ms, 1500ms, 2000ms, etc so that each time it fails we wait a bit longer but early on we can retry very fast. For now though I wouldn't worry about this just something to keep in mind.